### PR TITLE
dnscrypt-proxy 2.0.4

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Don't ship this yet; I'm not 100% done testing all of this. Just opening so ILZ can peek at it.

### CODE REVIEW

**Comment by ilovezfs** [comment]
Yeah not really eager to switch to this until Debian decides to bless what seems like a fork.

**Comment by DomT4** [comment]
It's the same person, no? Unless I missed something.

**Comment by ilovezfs** [comment]
Not sure how to interpret https://twitter.com/jedisct1/status/928942292202860544

**Comment by DomT4** [comment]
I think they dropped the v1 version because the codebase was a mess & created something that is a lot simpler to maintain, based on the Wiki. There are recent tweets referencing the new repo, like https://twitter.com/jedisct1/status/960898960352333824. It's a weird situation. I could always reply to the original tweet & ask I guess.

**Comment by ilovezfs** [comment]
Hmmm ... weird. Maybe he changed his mind?

**Comment by DomT4** [comment]
Hey @jedisct1, can you clear this up? Did you decide to retain ownership of the project?

**Comment by DomT4** [comment]
Always worth trying that. Most people seem to keep GitHub ping notifications turned on.

**Comment by ilovezfs** [comment]
so is minisign no longer a thing here

**Comment by DomT4** [comment]
It's been turned into a `go` library, which is vendored.

**Comment by ilovezfs** [comment]
and what happened to dnscrypt-update-resolvers

**Comment by DomT4** [comment]
Fetches & caches them itself now:

```
[sources]

  ## An example of a remote source

  [sources.'public-resolvers']
  url = 'https://download.dnscrypt.info/resolvers-list/v2/public-resolvers.md'
  cache_file = 'public-resolvers.md'
  minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
  refresh_delay = 72
  prefix = ''

  ## Another example source, with resolvers censoring some websites not approriate for children
  ## This is a subset of the `public-resolvers` list, so enabling both is useless

  #  [sources.'parental-control']
  #  url = 'https://download.dnscrypt.info/resolvers-list/v2/parental-control.md'
  #  cache_file = 'parental-control.md'
  #  minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'



## Optional, local, static list of additional servers
## Mostly useful for testing your own servers.
```

**Comment by DomT4** [comment]
The path there is now:
```
/usr/local/etc/public-resolvers.md
/usr/local/etc/public-resolvers.md.minisig
```
So we may wish to tweak/override that for consistency given we were using `pkgshare` before but I don't think it's a big deal to leave as-is unless I'm missing something.

**Comment by ilovezfs** [comment]
I mean what's the deal with updating it now. You've removed the post install altogether.

**Comment by DomT4** [comment]
It updates itself: `refresh_delay = 72`. See https://github.com/jedisct1/dnscrypt-proxy/issues/85. You can delete the file & restart the proxy to force a manual fetch if desired, or it'll just happen every 72 hours.

**Comment by ilovezfs** [comment]
Is there any use case here for running it undaemonized? I'm wondering if we should escape from sbin purgatory.

**Comment by DomT4** [comment]
I don't think so, personally. There's not a lot you'd want to interact with manually:

```
Usage of dnscrypt-proxy:
  -config string
    	Path to the configuration file (default "dnscrypt-proxy.toml")
  -json
    	output list as JSON
  -list
    	print the list of available resolvers for the enabled filters
  -list-all
    	print the complete list of available resolvers, ignoring filters
  -logfile string
    	Write logs to file
  -loglevel value
    	Log level (0-6) (default 2)
  -resolve string
    	resolve a name using system libraries
  -service string
    	Control the system service: ["start" "stop" "restart" "install" "uninstall"]
  -syslog
    	Send logs to the local system logger (Eventlog on Windows, syslog on Unix)
  -version
    	Prints current proxy version
```

Pretty much everything now is more or less tweak the config & restart the service.

File: Formula/dnscrypt-proxy.rb [hunk 1]
```diff
@@ -144,6 +79,7 @@ def plist; <<~EOS
   end
 
   test do
-    system "#{sbin}/dnscrypt-proxy", "--version"
+    output = shell_output("#{sbin}/dnscrypt-proxy -config #{etc}/dnscrypt-proxy.toml -list 2>&1")
```
Comment (ilovezfs) [inline Formula/dnscrypt-proxy.rb]: .................................................................................................

Comment (DomT4) [reply Formula/dnscrypt-proxy.rb]: You'd like a shorter line?

Comment (ilovezfs) [reply Formula/dnscrypt-proxy.rb]: You must have a secret decoder ring.

Comment (DomT4) [reply Formula/dnscrypt-proxy.rb]: It's my personal "Google Translate" plugin.

**Comment by DomT4** [comment]
Updated to 2.0.2 and emailed the Debian maintainer to ask whether they're planning to pick this up in the near future.

**Comment by DomT4** [comment]
```
Hi Dominyk,

I'm intending to but v2 is going to be a bit of work given the Go
rewrite, unpackaged dependencies and different config file format. I'm
hoping to find some time this weekend to work on it.

* Dominyk Tiller (me@gmail.com) wrote:
> Hi Eric,
>
> I wondered whether Debian was planning to pick up the v2 rewrite of
> dnscrypt-proxy? I noticed the package tracker had flagged there being an
> issue checking for a new upstream version and wondered whether that was
> the issue with it not being considered for sid yet or whether there was
> another blocker.
>
> Upstream seem to have nuked the previous website in favour of simply
> hosting all the releases on GitHub
> (https://github.com/jedisct1/dnscrypt-proxy/releases).
>
> Thanks,
>
> Dominyk
```

**Comment by ilovezfs** [comment]
>Don't ship this yet; I'm not 100% done testing all of this.

Still in effect or are we ready?

**Comment by DomT4** [comment]
There's another update I'll push, but after that we're good to go if you're happy.

**Comment by DomT4** [comment]
Thanks Joe. I see we got a bottle that works in all prefixes out of it too, which is a nice bonus.

**Pull request closed without merging**